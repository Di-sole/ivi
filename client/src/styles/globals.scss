body {
    font-family: Arial, Helvetica, Helvetica Neue, FreeSans, sans-serif;
    background-color: #100e19;
}

body.modal-active {
    height: 100vh;
    overflow-y: hidden;
}

.container {
    max-width: 1300px;
    margin: 0 auto;
    padding: 0 24px;
}