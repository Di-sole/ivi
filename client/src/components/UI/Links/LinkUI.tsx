import { ReactNode } from 'react';
import styles from './LinkUI.module.scss';

interface LinkUIProps {
    children: ReactNode;
    linkTo?: string;
    shape: 'rectangular' | 'square' | 'round';
}

const LinkUI = ({ children, linkTo, shape }: LinkUIProps) => {
    const linkFormClass =
        shape === 'round'
            ? styles.link__wrapper_round
            : shape === 'square'
            ? styles.link__wrapper_square
            : styles.link__wrapper_rectangular;

    return (
        <a
            href={linkTo}
            className={[styles.link__wrapper, linkFormClass].join(' ')}
            target="blank"
            role="link-to-media"
        >
            <div className={styles.link__content}>{children}</div>
        </a>
    );
};

export default LinkUI;
