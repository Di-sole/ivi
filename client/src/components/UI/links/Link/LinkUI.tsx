import { ReactNode } from 'react';
import Link from 'next/link';
import styles from './LinkUI.module.scss';

interface LinkUIProps {
    children: ReactNode;
    href: string;
    shape: 'rectangular' | 'square' | 'round';
    className?: string;
}

const LinkUI = ({ children, href, shape, className }: LinkUIProps) => {
    const linkFormClass =
        shape === 'round'
            ? styles.container__wrapper_round
            : shape === 'square'
            ? styles.container__wrapper_square
            : styles.container__wrapper_rectangular;

    return (
        <Link
            href={href}
            className={[styles.container__wrapper, linkFormClass, className].join(' ')}
            target="blank"
            role="link-to-media"
        >
            <div className={styles.container__content}>{children}</div>
        </Link>
    );
};

export default LinkUI;
