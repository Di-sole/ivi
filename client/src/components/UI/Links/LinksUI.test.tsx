import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Image from 'next/image';
import LinkUI from './LinkUI';
import styles from './LinkUI.module.scss';
import devicesIcon from '../../../../public/icons/link_icons/devices_all.svg';

describe('LinksUI SNAPSHOT TESTS', () => {
    test('Round link should not be changed', () => {
        render(
            <LinkUI linkTo="https://t.me/official_iviru" shape="round">
                <img
                    src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_telegram.svg"
                    alt="Логотип Telegram"
                />
            </LinkUI>,
        );
        const link = screen.getByRole('link-to-media');

        expect(link).toMatchSnapshot();
    });

    test('Rectangular link should not be changed', () => {
        render(
            <LinkUI linkTo="#" shape="rectangular">
                <div className={styles.text__single}>
                    <Image src={devicesIcon} height="20" width="20" alt="Devices logo" />
                    Все устройства
                </div>
            </LinkUI>,
        );
        const link = screen.getByRole('link-to-media');

        expect(link).toMatchSnapshot();
    });
});
