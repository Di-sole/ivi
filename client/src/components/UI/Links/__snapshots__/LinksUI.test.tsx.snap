// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinksUI SNAPSHOT TESTS Rectangular link should not be changed 1`] = `
<a
  class="link__wrapper link__wrapper_rectangular"
  href="#"
  role="link-to-media"
  target="blank"
>
  <div
    class="link__content"
  >
    <div
      class="text__single"
    >
      <img
        alt="Devices logo"
        data-nimg="1"
        decoding="async"
        height="20"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
        style="color: transparent;"
        width="20"
      />
      Все устройства
    </div>
  </div>
</a>
`;

exports[`LinksUI SNAPSHOT TESTS Round link should not be changed 1`] = `
<a
  class="link__wrapper link__wrapper_round"
  href="https://t.me/official_iviru"
  role="link-to-media"
  target="blank"
>
  <div
    class="link__content"
  >
    <img
      alt="Логотип Telegram"
      src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_telegram.svg"
    />
  </div>
</a>
`;
